### PR TITLE
KlayGE: Fix compiling in OSX/iOS/Android

### PR DESCRIPTION
**Pull request by qiankanglai**
Fix:
- ignore unused parameter in clang
- typo in KFL/Config.hpp
- add namespace if needed
- adjust tool path in CMakeLists.txt for iOS/Android
- remove unused variables (I think clang is more strict than msvc or gcc. If one variable is only used in construction, it is in fact useless)
- some overrides

TODO:
`IntegratedBRDF.dds/OceanDisplacement.dds` do not exist until demos are first run. If we build iOS/Android without running these examples in host platform, we will get errors copying media files.

### CODE REVIEW

File: KlayGE/Core/Include/KlayGE/Input.hpp [hunk 1]
```diff
@@ -38,7 +38,6 @@
 #include <KFL/Vector.hpp>
 #include <KFL/Timer.hpp>
 
-#include <boost/container/flat_map.hpp>
```
Comment (gongminmin) [inline KlayGE/Core/Include/KlayGE/Input.hpp]: Please keep this line and move the -Wunused-parameter part here.

File: KlayGE/Core/Src/AppLayer/WindowDarwin.mm [hunk 2]
```diff
@@ -162,7 +164,9 @@ static void RegisterApp()
 				a_size = 2;
 				break;
 				
-			default:
+            default:
```
Comment (gongminmin) [inline KlayGE/Core/Src/AppLayer/WindowDarwin.mm]: Use tab instead of space.

File: KlayGE/Core/Src/Render/SATPostProcess.cpp [hunk 3]
```diff
@@ -31,9 +31,9 @@ namespace KlayGE
 					std::vector<std::string>(),
 					std::vector<std::string>(1, "src_tex"),
 					std::vector<std::string>(1, "output"),
-					tech),
-				dir_(dir)
+					tech)
 	{
+		UNREF_PARAM(dir);
```
Comment (gongminmin) [inline KlayGE/Core/Src/Render/SATPostProcess.cpp]: We can remove "dir" parameter. Seems it's not used at all.

File: KlayGE/Plugins/Include/KlayGE/OpenGL/OGLRenderView.hpp [hunk 4]
```diff
@@ -225,7 +225,6 @@ namespace KlayGE
 		GLenum target_type_;
 		int array_index_;
 		int level_;
-		uint32_t sample_count_, sample_quality_;
```
Comment (gongminmin) [inline KlayGE/Plugins/Include/KlayGE/OpenGL/OGLRenderView.hpp]: Please keep those two member variables, and use UNREF_PARAM in constructor on them. Render-to-MSAA-texture will be implemented in the future.

File: KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp [hunk 5]
```diff
@@ -99,7 +99,6 @@ namespace KlayGE
 
 	private:
 		OGLESTexture1D& texture_1d_;
-		int array_index_;
```
Comment (gongminmin) [inline KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp]: Please keep array_index, and use UNREF_PARAM in constructor on it. Texture array will be implemented very soon.

File: KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp [hunk 6]
```diff
@@ -120,7 +119,6 @@ namespace KlayGE
 
 	private:
 		OGLESTexture2D& texture_2d_;
-		int array_index_;
```
Comment (gongminmin) [inline KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp]: Same as last one. Texture array will be implemented very soon.

File: KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp [hunk 7]
```diff
@@ -193,9 +191,7 @@ namespace KlayGE
 
 	private:
 		GLenum target_type_;
-		int array_index_;
```
Comment (gongminmin) [inline KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp]: Please keep array_index.

File: KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp [hunk 8]
```diff
@@ -193,9 +191,7 @@ namespace KlayGE
 
 	private:
 		GLenum target_type_;
-		int array_index_;
 		int level_;
-		uint32_t sample_count_, sample_quality_;
```
Comment (gongminmin) [inline KlayGE/Plugins/Include/KlayGE/OpenGLES/OGLESRenderView.hpp]: Please sample count and quality.

File: KlayGE/Samples/src/CausticsMap/CausticsMap.cpp [hunk 9]
```diff
@@ -113,7 +113,7 @@ namespace
 			caustics_map_ = caustics_map;
 		}
 
-		void Pass(uint32_t pass)
+		void Pass(PassType pass)
```
Comment (gongminmin) [inline KlayGE/Samples/src/CausticsMap/CausticsMap.cpp]: We'd better use a new function name, such as CausticsPass. It's not exactly a override case.

File: KlayGE/Tools/src/HDRCompressor/HDRCompressor.cpp [hunk 10]
```diff
@@ -453,7 +453,7 @@ int main(int argc, char* argv[])
 
 	if (argc < 2)
 	{
-		cout << "ʹ�÷���: HDRCompressor xxx.dds [R16 | R16F] [BC5 | BC3]" << endl;
+		cout << "usage: HDRCompressor xxx.dds [R16 | R16F] [BC5 | BC3]" << endl;
```
Comment (gongminmin) [inline KlayGE/Tools/src/HDRCompressor/HDRCompressor.cpp]: "Usage"

File: blib_util.py [hunk 11]
```diff
@@ -104,7 +104,9 @@ def __init__(self, compiler, archs, cfg):
 				if space_place != -1:
 					android_ver = target_platform[space_place + 1:]
 					target_platform = target_platform[0:space_place]
-					if "5.0" == android_ver:
+					if "5.1" == android_ver:
```
Comment (gongminmin) [inline blib_util.py]: Please move android 5.1 to a separated commit.

File: glloader/src/utils.cpp [hunk 12]
```diff
@@ -182,28 +181,28 @@ namespace glloader
...
-		if (ogl_dll != NULL)
-		{
-			gl_dlls_.push_back(ogl_dll);
-		}
-		ogl_dll = ::dlopen("libEGL.dylib", RTLD_LAZY);
-		if (ogl_dll != NULL)
-		{
-			gl_dlls_.push_back(ogl_dll);
-		}
+			// http://forum.imgtec.com/discussion/comment/18323#Comment_18323
```
Comment (gongminmin) [inline glloader/src/utils.cpp]: An extra tab?

**Comment by qiankanglai** [comment]
updated Pull Request, according to comments

File: KlayGE/Core/Include/KlayGE/Input.hpp [hunk 13]
```diff
@@ -53,6 +57,9 @@
 #elif defined(KLAYGE_COMPILER_GCC)
 #pragma GCC diagnostic pop
 #endif
+#if defined(KLAYGE_COMPILER_CLANG)
```
Comment (gongminmin) [inline KlayGE/Core/Include/KlayGE/Input.hpp]: Those 3 lines should be moved to the line after #include <boost/container/flat_map.hpp>

**Comment by qiankanglai** [comment]
Input.hpp updated

**Pull request merged**